src/v/cluster/members_backend.cc

```diff
@@ -854,25 +974,25 @@ void members_backend::stop_node_decommissioning(model::node_id id) {
     }
     // remove all pending decommissioned updates for this node
     std::erase_if(_updates, [id](update_meta& meta) {
-        return meta.update.id == id
-               && meta.update.type
+        return meta.update && meta.update->id == id
+               && meta.update->type
                     == members_manager::node_update_type::decommissioned;
     });
 }
 
 void members_backend::stop_node_addition(model::node_id id) {
     // remove all pending added updates for current node
     std::erase_if(_updates, [id](update_meta& meta) {
-        return meta.update.id == id
-               && meta.update.type == members_manager::node_update_type::added;
+        return !meta.update
```

Comment:
nit: probably it makes sense to remove manual rebalance update after we decommission the node, but the function name starts looking kind of wrong.

Reply:
will rename